module/mempool/stdmap/pending_receipts.go

```diff
@@ -0,0 +1,109 @@
+package stdmap
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// PendingReceipts stores pending receipts indexed by the id.
+// It also maintains a secondary index on the previous result id, which is unique,
+// in order to allow to find a receipt by the previous result id.
+type PendingReceipts struct {
```

Comment:
I am concerned about this mempool implementation:
* it stores a _single_ receipt by its parent result
* However, there will be _multiple_ receipts (from different ENs) for the same parent. Keeping one is not sufficient, as we want to reward _all_ ENs, who produce a receipt in time. Hence, we have to keep all receipts.

Reply:
yeah, makes sense